oracle/market_mapper.go

```diff
@@ -3,13 +3,61 @@ package oracle
 import (
 	"context"
 	"encoding/json"
+	"fmt"
 	"os"
 	"strings"
 	"time"
 
 	"go.uber.org/zap"
+
+	mmtypes "github.com/skip-mev/connect/v2/x/marketmap/types"
 )
 
+// IsMarketMapValidUpdated checks if the given MarketMapResponse is an update to the existing MarketMap.
+// - returns an error if the market map is fully invalid or the response is invalid
+// - returns false if the market map is not updated
+// - returns true and the new market map if the new market map is updated and valid.
+func (o *OracleImpl) IsMarketMapValidUpdated(resp *mmtypes.MarketMapResponse) (mmtypes.MarketMap, bool, error) {
+	if resp == nil {
```

Comment:
Looks like we didn't check for result.Value to be nil before, are we checking it now because we are not sure if it can be nil, or just because we are exposing this as a function?

Reply:
It should not be `nil` if it is returned from `x/marketmap`, but since the client is generic, its good to have nil checks in place since this is now a function